frontend/src/pages/ExperimentDetails.tsx

```diff
@@ -267,20 +269,34 @@ class ExperimentDetails extends Page<{}, ExperimentDetailsState> {
         pageTitleTooltip: pageTitle,
       });
 
-      // TODO: get ALL jobs in the experiment
-      const recurringRuns = await Apis.jobServiceApi.listJobs(
-        undefined,
-        100,
-        '',
-        ApiResourceType.EXPERIMENT.toString(),
-        experimentId,
-      );
-      const activeRecurringRunsCount =
-        (recurringRuns.jobs || []).filter(j => j.enabled === true).length;
-      this.setState(
-        { activeRecurringRunsCount, experiment },
-        () => this._runlistRef.current && this._runlistRef.current.refresh()
-      );
+      let activeRecurringRunsCount = -1;
+
+      // Fetch this experiment's jobs
+      try {
+        // TODO: get ALL jobs in the experiment
+        const recurringRuns = await Apis.jobServiceApi.listJobs(
+          undefined,
+          100,
+          '',
+          ApiResourceType.EXPERIMENT.toString(),
+          experimentId,
+        );
+        activeRecurringRunsCount =
+          (recurringRuns.jobs || []).filter(j => j.enabled === true).length;
+
+      } catch (err) {
+        await this.showPageError(
+          `Error: failed to retrieve recurring runs for experiment: ${experimentId}.`, err);
+        logger.error(`Error fetching recurring runs for experiment: ${experimentId}`, err);
+      }
+
+      this.setStateSafe({ activeRecurringRunsCount, experiment });
+      
+      // TODO: maybe move outside of try-catch
```

Comment:
I think there's no harm in that, even if an error happens above it in the try/catch.

Reply:
Done